server.c

```diff
@@ -163,36 +167,109 @@ void *ClientHandler(void *cli)
 			*res = 0;
 			return (void *)res;
 		}
-	}
-}
 
-int sendResponse(int sockfd, const char *response)
-{
-	return send(sockfd, response, strlen(response), 0);
-}
-
-// te funkcja jest stara i moze nie miec sensu
-char *GetFilenameFromRequest(char *request)
-{
-	printf("\n");
-	char *file_name = strchr(request, ' ');
-	printf("%s", file_name);
-	return file_name + 1;
-}
+		//obsluga komendy LIST
+		if (strcmp(request_token, "LIST") == 0)
+		{
+			if (client.data_socket > 0)
+			{
+				sendResponse(client.socket, "150 Here comes the directory listing. \r\n");
+				request_token = strtok(NULL, " \r\n");
+				if (request_token == NULL)
+				{
+					char cwd[256];
+					getcwd(cwd, sizeof(cwd));
+					request_token = &cwd[0];
+				}
+				DIR *dir = opendir(request_token);
+				char path[256];
+				struct stat buf;
+				struct tm *t;
+				char timebuff[80];
+				time_t time;
+				if (dir != NULL)
+				{
+					struct dirent *file;
+					while ((file = readdir(dir)))
+					{
+						if ((strcmp(file->d_name, ".") != 0) && (strcmp(file->d_name, "..") != 0))
+						{
+							strcpy(path, request_token);
+							strcat(path, "/");
+							strcat(path, file->d_name);
+							if (stat(path, &buf) != -1)
+							{
+								time = buf.st_mtime;
+								t = localtime(&time);
+								strftime(timebuff, 80, "%b %d %H:%M", t);
+								sprintf(client_response, "%c%s %5d %4d %4d %8d %s %s \r\n",
```

Comment:
Testowałaś to? Chyba nie tak się formatuje stringa w C. Powinno być chyba po prostu `%d` a nie `%<n>d`. U mnie kompilator wyrzuca warningi.

Reply:
dobra sorry, tu błędy były bo tam musisz `%ld` dać w pierwszym i ostatnim